typedload/dataloader.py

```diff
@@ -815,6 +817,14 @@ def _datetimeload(l: Loader, value: Any, type_) -> Union[datetime.date, datetime
         raise TypedloadValueError(str(e), type_=type_, value=value)
 
 
+def _patternload(l: Loader, value: Any, type_) -> re.Pattern:
+    if hasattr(type, "__args__"):
+        (input_type,) = type_.__args__
+        if input_type in {bytes, str} and type(value) != input_type:
+            raise TypedloadValueError('Got %s of type %s, expected %s' % (repr(value), tname(type(value)), tname(type_)), value=value, type_=type_)
+    return re.compile(value)
```

Comment:
The re.compile might raise an exception. It seems it can be a TypeError or a re.error.

They have to be wrapped into typedload exceptions, to be able to later inspect where in the datatree the issue occurred.